### DISCUSSION TITLE
How to disable Vercel Insight when hosting NextJS on vercel

### DISCUSSION THREAD

**Discussion by heinthanth** [vercel/vercel#8575] [Help] [answered]
I have a NextJS app hosted on vercel. I tried to look grep `/v1/vitals` inside built `.next` dir but didn't find.
So, I believe that insight report code is not in local machine.

But when hosting on vercel, I found some API requests to `https://vitals.vercel-insights.com/v1/vitals`.

<img width="955" alt="Xnip2022-09-18_21-53-55" src="https://user-images.githubusercontent.com/43804263/190914938-a64f6893-e798-4bb1-bb28-fcbc2a554df9.png">

How can I disable that insight report or analytics?

**Comment by heinthanth** [comment]
Although I disable it ( after enabling analytics ) in setting dashboard.

But, I'm building NextJS on Github CI. So currently, what I did is

```shell
vercel pull
# edit projects.json and set analyticsId to null
vercel build
```

It works. But project setting json should remove tracking ID if disabled in project setting UI.

**Comment by amyegan** [comment, marked as answer]
Hi @heinthanth! You can disable Analytics from the Dashboard. Go to your project dashboard -> analytics -> click the `...` button on the right-hand side to open a menu -> click "Disable Analytics". The next time you deploy the site, you will no longer see the `vitals` request.

<img width="833" alt="Screen Shot 2022-09-19 at 6 14 37 PM" src="https://user-images.githubusercontent.com/8247105/191129344-c417dd08-eaa9-48a6-9752-c3b3992cfee4.png">

**Reply by heinthanth** [reply]
No, unfortunately, it sent vitals even analytics is disabled.

PS: I'm building NextJs in GitHub action using vervet build command.

Vervet pull command produce a project json file with AnalyticsID.

**Reply by amyegan** [reply]
I'm sorry you're still facing this problem. Do you have any other plugins or API calls set up that could be related to this? Please share a [minimal reproducible example](https://vercel.com/support/articles/creating-a-minimal-reproducible-example) (or your repository if it is public) so I can help you identify what is causing the [analytics request](https://vercel.com/docs/concepts/analytics/api) in your app.

**Reply by heinthanth** [reply]
@amyegan 

Here is the repo: https://github.com/heinthanth/dotspace
And here is the hosted site: https://heinthantdotspace.vercel.app/

For now, requests are disabled because I disabled it with weird method ( edited `.vercel/projects.json` )

https://github.com/heinthanth/dotspace/blob/main/.github/workflows/test-and-launch.yaml#L29-L30
https://github.com/heinthanth/dotspace/blob/main/src/utilities/disable-tracking.mjs#L10

That means although I disabled "ANALYTICS" in dashboard, `vercel pull` command produce the `projects.json` with `analyticsId` ( which should be `null` if disabled ).

Thanks in advance!

**Reply by amyegan** [reply]
Thanks for sharing this! It looks like `next/core-web-vitals` is included in your `.eslintrc.cjs`. I believe that is the source of the `analyticsId` and `v1/vitals` requests you're seeing. I found it here: https://github.com/heinthanth/dotspace/blob/3880952206452436b6e45a1b6cb2c108dd607efe/.eslintrc.cjs#L13

**Reply by heinthanth** [reply]
But `eslint` in a linter and not supposed to be included in production build of NextJS. And it won't have knowledge of `analyticsId` from `projects.json`.

And this is build from a commit ( before I enable Analytics From Dashboard ):
https://heinthantdotspace-dcovstqsm-heinthanth.vercel.app/
https://github.com/heinthanth/dotspace/tree/2ad73f223368367fd5cc13b0be8cb213116a957c

Since before that commit, `next/core-web-vitals` and `eslint` exists and no vital requests until enable Analytics.
But after I disable Analytics, there's still vital requests.

Is that related to project not building on vercel ( since I'm building it on Github Action using `vercel pull` and `vercel build` )?

**Reply by heinthanth** [reply]
So, what I'm guessing is "Disable Analytics button in Dashboard is not working or have bugs in certain condition ( if it happening only to me )"

**Reply by heinthanth** [reply]
More Evidence:

This is normal NextJS build using `npm run build`.

<img width="1020" alt="image" src="https://user-images.githubusercontent.com/43804263/191516176-7d3c076c-5bbd-48e3-8acd-60e6c719787c.png">

And when u grab for vital API URL, `source map file` is matched. That means vital report code is not in the app.

<img width="1011" alt="image" src="https://user-images.githubusercontent.com/43804263/191516414-799cb0c2-8433-4e5f-b06a-5caf5074a5e0.png">


But when building using `vercel pull` and `vercel build`.

<img width="1014" alt="image" src="https://user-images.githubusercontent.com/43804263/191516727-868b013d-1163-4d4a-be62-87813abfc7a5.png">
 
 You can the message "Analytics is enabled"
 
<img width="1009" alt="image" src="https://user-images.githubusercontent.com/43804263/191516791-2ad3d7b2-3cff-4752-98ab-1c61eca41af9.png">

And ".next/static/chunks/main-39722929a634dbbb.js" is matched here.

<img width="1024" alt="image" src="https://user-images.githubusercontent.com/43804263/191516870-8b467500-13a5-4fbb-a381-4d24f60b4288.png">

So, I think it does not depends on my code. I think something wrong with vercel project setting. Here's generated `.vercel/projects.json` file.

<img width="684" alt="image" src="https://user-images.githubusercontent.com/43804263/191517201-9aa03fe3-f89e-4ff6-999f-093c34170d42.png">

**Reply by amyegan** [reply]
Thank you for sharing all of these details. A fix is on the way!  https://github.com/vercel/vercel/pull/8607

**Reply by heinthanth** [reply]
https://github.com/vercel/vercel/pull/8607 fixed the issue. Thank you so much for the hard work!

**Reply by kkggg55** [reply]
I removed analyticsId and deployed my app, but vitals is still sending requests back and forth. What's the solutioun?

**Comment by monolithed** [comment]
@amyegan, 
<img width="1156" alt="Screenshot 2023-09-10 at 19 02 51" src="https://github.com/vercel/vercel/assets/803674/f1e1996e-6ae7-485b-9b18-e0199bffe438">

Is It normal to have duplicate requests to http://vitals.vercel-insights.com? Sometimes these requests take 1-3 seconds to load. Could they affect the page load speed? I'm asking because my static pages take 3-4 seconds to load.
